### ISSUE TITLE
calling aclose() on a subscription doesn't cancel task

### ISSUE THREAD

**Issue by dfee** [graphql-python/graphql-core-next#9]
I put together a new version of `graphql-ws` specifically designed for `graphql-core-next`, and one problem I've run into is that calling `aclose()` on an async generator doesn't cancel whatever is waiting on its `__anext__()` (https://bugs.python.org/issue28721)

The result is that subscriptions in graphql-core-ws aren't really cancellable, and instead emit the following error:

```python
Task was destroyed but it is pending!
task: <Task pending coro=<<async_generator_asend without __name__>()> wait_for=<Future pending cb=[<TaskWakeupMethWrapper object at 0x10ed07d68>()]> cb=[_wait.<locals>._on_completion() at /Users/dfee/.pyenv/versions/3.7.0/lib/python3.7/asyncio/tasks.py:436]>
^
```

BTW, you can check out that repo (pre-publish) at https://github.com/dfee/graphql-next-ws

**Comment by Cito** [comment]
That's interesting. I have been also working on a new version of `graphql-ws` recently (not published yet). It includes the functionality from `graphql-subscriptions` and I want to add all the tests and the client code from `graphql-ws` as well. Maybe we can join our efforts.

**Comment by dfee** [comment]
Is this the [`graphql-subscriptions`](https://github.com/apollographql/graphql-subscriptions) you're referencing?

I'm totally open to joining efforts. I'd like to sprinkle into that repository some brief documentation regarding usage, and then migrate the project to this team and I'd be happy to grant you write-access (@syrusakbary, thoughts?).

Thanks for your work with `graphql-core-next`!

**Comment by Cito** [comment]
Yes, that's the `graphql-subscriptions` I have included in my version of `graphql-ws`.  In my implementation of `with_filter` I'm using a similar mechanismus as you recently suggested to allow closing a filtered async iterable while it's running.

**Comment by dfee** [comment]
Cool. Yeah, I'd love to get that merged in with this team and make it a shared project (I've reached out to @syrusakbary via email...).

**Comment by dfee** [comment]
There were some worthwhile additions to https://bugs.python.org/issue34730 that might be worth reading as we implement async gens (but I don't think that's a problem with async iters).

**Comment by Cito** [comment]
Closing this since I see no more actionable items for graphql-core-next. graphql-next-ws is a separate issue (still on my todo list).
